Added mysql-based functional tests

asadoughi opened the pull request:
Run with `tox -e mysql`. It's configured to connect to
"mysql://root@localhost/neutron".

RM11693

asadoughi commented:
Oh, apparently travis-ci ran that flawlessly without needing to create a database. I would have added 
```
before_script:
  - mysql -e 'create database myapp_test;'
```
as mentioned here http://docs.travis-ci.com/user/database-setup/#MySQL but it seems like we don't need it. Cool.

jmeridth commented:
@asadoughi that is pretty cool.  TIL

insequent commented on tox.ini:
```diff
@@ -32,7 +32,20 @@ setenv = VIRTUAL_ENV={envdir}
+         NOSE_OPENSTACK_STDOUT=1
+         NO_EVENTLET=1
+deps = -r{toxinidir}/requirements.txt
+       -r{toxinidir}/test-requirements.txt
```
So I was working with Tox a little over the weekend, and I *believe* that these variables are not needed because they're already defined in the default testenv, and you only have to include whatever you want to override. That is, unless I'm misreading http://tox.readthedocs.org/en/latest/config.html

insequent commented:
This is super cool. :snowman:

asadoughi commented:
Updated connection string to "mysql://root@localhost/quark_functional_tests" so it doesn't drop regular "neutron" database, also changed tox.ini to reflect @insequent's comment.

asadoughi commented:
Should I make the "mysql" test environment a default?

asadoughi commented:
...I think I will.

insequent commented:
Be bold. LGTM :+1:

jmeridth commented:
py26 drop in #358 causes this to need rebase.  Ready to merge once done.

asadoughi commented:
Rebased
